sklearn/svm/_base.py

```diff
@@ -677,14 +680,18 @@ def n_support_(self):
             check_is_fitted(self)
         except NotFittedError:
             raise AttributeError
+        return self._n_support
 
-        svm_type = LIBSVM_IMPL.index(self._impl)
-        if svm_type in (0, 1):
-            return self._n_support
-        else:
-            # SVR and OneClass
-            # _n_support has size 2, we make it size 1
-            return np.array([self._n_support[0]])
+    @property
+    def _class_weight(self):
+        """Weights per class"""
+        # Avoid using self.class_weight_ in SVR, NuSVR, OneClass due
+        # due to deprecation
+        return (
+            self.class_weight_
+            if LIBSVM_IMPL.index(self._impl) in (0, 1)
+            else np.empty(0)
+        )
```

Comment:
I prefer to use `hasattr` here:

```suggestion
        # Class weights are defined for classifiers during
        # fit, otherwise it is empty
        if hasattr(self, "class_weight_"):
            return self.class_weight_
        return np.empty(0)
```

This way we do not couple `LIBSVM_IMPL.index(self._impl) in (0, 1)` with class weights. The logic becomes "If there are class weights, use them, otherwise do not".